mvcc/kvstore.go

```diff
@@ -379,6 +379,26 @@ func (s *store) restore() error {
 	tx := s.b.BatchTx()
 	tx.Lock()
 
+	// new keys introduced in 3.5.
```

Comment:
You picked a pretty good place to add logic for handling db file. I think that I even missed in the original downgrade design that only migrates storage when bootstrapping etcd. Added https://github.com/etcd-io/etcd/issues/17361 to improve it.

However the validation you added would not work. What happens if during downgrade v3.4 member gets a snapshot from leader on v3.5? We need to reimplement parts of migrate code to detect storage version and downgrade it to v3.4, so we remove "term" and "confState" if encountered.

Reply:
done